modbus/modbus.go

```diff
@@ -0,0 +1,266 @@
+package modbus
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+
+	"github.com/goburrow/modbus"
+	"github.com/goburrow/serial"
+)
+
+type DataPoint string
+
+const (
+	Coil            DataPoint = "coil"
+	HoldingRegister DataPoint = "h_register"
+	InputRegister   DataPoint = "i_register"
+	Register        DataPoint = "register"
+	Discrete        DataPoint = "discrete"
+	FIFO            DataPoint = "fifo"
+)
+
+var (
+	errInvalidInput    = errors.New("invalid input type")
+	errUnsupportedRead = errors.New("invalid iotype for Write method: register")
+)
+
+type ModbusService interface {
+	// Read gets data from modbus.
+	Read(address, quantity uint16, iotype DataPoint) ([]byte, error)
+	// Write writes a value/s on Modbus.
+	Write(address, quantity uint16, value interface{}, iotype DataPoint) ([]byte, error)
+	// Close closes the modbus connection.
+	Close() error
+}
+
+var _ ModbusService = (*modbusService)(nil)
+
+// adapterService provides methods for reading and writing data on Modbus.
+type modbusService struct {
```

Comment:
```suggestion
type service struct {
```

Also fix the comment